compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -330,7 +333,15 @@ impl<'f> LoopInvariantContext<'f> {
             induction_vars.get(&binary.lhs),
             induction_vars.get(&binary.rhs),
         ) {
-            (Some((lhs, _)), None, None, Some((_, upper_bound))) => (lhs, *upper_bound),
+            (Some((lhs, _)), None, None, Some((lower_bound, upper_bound))) => {
+                if matches!(binary.operator, BinaryOp::Div | BinaryOp::Mod) {
+                    // If we have a Div/Mod operation we want to make sure that the
+                    // lower bound is not zero.
+                    (lhs, *lower_bound)
```

Comment:
Can edit the comment below on line 357 too.

"We evaluate this expression using the upper bounds (or lower in the case of div/mod) of its inputs to check..."